troopers.sh

```diff
@@ -1,76 +1,89 @@
 #!/bin/bash
 
-prefix="`dirname $0`/"          # prefix should be with "/" in the end
+trap "cleanup" EXIT
+
 login=$1                        # 1st argument of cli
 password=$2                     # 2nd argument of cli
-curl_opt="-s -b ${prefix}cookie.$login -c ${prefix}cookie.$login"
-exit_cycle=0
-selected_enemy=1                # if you want fight with specific avatar
 friend=$3
-if [ -z "$friend" ]
-then
-  friend="roushet"             # avatar's name
-fi
+site="http://$login.minitroopers.com"
+cookie_file="$(mktemp -t "$login.XXXXXX" --suffix='.cookie')"
+curl="curl --silent --cookie "$cookie_file""
+egrep="grep --extended-regexp --only-matching"
 
 # Check for "raids"
-function check {
-    message=`egrep "(Another|Shortage)" ${prefix}index`
-    if [ "$message" == "Shortage" ] || [ -z "$message" ]
-    then
-        exit_cycle=1
-        mission
-    fi
-}
-
-# Get money
-function getmoney {
-    money=`grep money ${prefix}index -A1|tail -n1`
-    echo "$login has earned $money coins"
+function hasRecruits {
+    $curl "$site/hq" | grep "Another" > /dev/null
+    return $?
 }
 
 # Make 3 "mission" tasks
 function mission {
-    mission_key=`egrep -o -e "chk=[A-Za-z0-9]{6}" ${prefix}index |tail -n1`
     for i in {1..3}
     do
-        curl $curl_opt http://$login.minitroopers.com/b/mission?$mission_key
+        $curl "$site/b/mission?chk=$chk"
     done
-    fight
 }
 
-# Make "fight" tasks
-function fight {
-    for i in {1..3}
+function getCheck {
+    local key="$($egrep --regexp='keyy6:[A-Za-z0-9]{6}y[0-9]:' $1)"
+    echo ${key:6:6}
+}
+
+function getFightKey {
+    $curl --cookie-jar "$cookie_file" "$site/b/opp" \
+        | $egrep --regexp='opp=[0-9]{5,7};chk=[A-Za-z0-9]{6}' \
+        | head --lines=1
```

Comment:
The only exception to what I mentioned on cookies, is when getting the list armies you can attack.
But that's only a problem when you want to attack armies at random.